gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2395,8 +2629,17 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
         else if( nType == NC_INT )
             nodata.nVal = NC_FILL_INT;
 
-        status = nc_def_var(m_nLayerCDFId, pszVarName, nType, 1, &nMainDimId,
-                            &nVarID);
+        if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation